### PR TITLE
[wip] add server xDS support

### PR DESCRIPTION
**Pull request by joeljeske**
Basic integration for server support. 

Tests to add: 
- Validate if the various serverOptions are compatible when using xDS server or there are runtime errors
- Validate basic static xDS bootstrap config reading

### CODE REVIEW

File: pkg/proto/configuration/grpc/grpc.proto [hunk 1]
```diff
@@ -164,8 +164,8 @@ message ServerConfiguration {
   // inside directories that have access controls set up properly.
   repeated string listen_paths = 2;
 
-  // TLS configuration. TLS is not enabled when left unset.
-  buildbarn.configuration.tls.ServerConfiguration tls = 3;
+  // was TLS, moved tl TLS option in transport_credentials.
+  reserved 3;
```
Comment (EdSchouten) [inline pkg/proto/configuration/grpc/grpc.proto:168]: No need to mark 3 as reserved. Just let `tls` below use field number 3 again. With Protobuf it's safe to upgrade a single field to a `oneof`, provided you preserve the field number.

**Comment by joeljeske** [comment]
TL;DR Closing this in favor of #149 

I initially started down this path of integrating into xDS with the goal of supporting certificate rotation from cert files on disk. In reading #120 thought it best to use xDS to support this use case. I did not have plans to actually integrate with a running xDS server, but hoping use some of the client-only grpc golang features of xDS. I have decided that this is not a good approach for the following reasons. 

First, I've found that xDS integration with go-gRPC does not appear to function without the use of an accessible xDS server to manage configuration. I was originally under the impression that after integration with go-grpc/xds, we would be able to specify a xDS bootstrap file that is static on only read from disk without the need for a xDS server. Whereas the xDS spec does allow for a file_watcher[ certificate provider](https://github.com/grpc/proposal/blob/master/A29-xds-tls-security.md#file_watcher-certificate-provider), there must still be an xDS server that requests clients to use such a provisioned certificate.

Second, I have found that there are a variety of expectations made on an application that claims to support xDS. If Buildbarn was xDS compatible, one would expect that they could configure the following bits on their xDS server: Service Discovery, Route Authorization, Load Balancing, Retry/Backoff. I will not be using any of those features, and am not the right person to integrate those bits into Buildbarn. Obviously there is significant overlap with this feature set with the existing configuration of Buildbarn. It is probably not desirable to support this alternate strategy without deeper analysis.
